### ISSUE TITLE
Configure: accept --api=3 and --api=3.x options

### ISSUE THREAD

**Issue by mspncp** [openssl/openssl#10166]
Currently the configure command only accepts `--api=3.0.0`, although the third digit is only a patch level which does not permit api changes. It should be `--api=3.0` instead. (This is the bug report).

In fact, it would be convenient, if missing trailing version numbers would be automatically set to 0, i.e., if one could specify `--api=3` and this would be expanded to `--api=3.0` (this is the feature request).

In fact, do we actuall support deprecations at minor revision boundaries? If not, then we should allow only `--api=n` for n>=3.

```
msp@msppc:~/openssl/openssl$ perl Configure linux-x86_64 --api=3.0.0
Configuring OpenSSL version 3.0.0-dev for target linux-x86_64
Using os-specific seed configuration
Creating configdata.pm
Running configdata.pm
Creating Makefile
...
```

```
msp@msppc:~/openssl/openssl$ perl Configure linux-x86_64 --api=3.0

Failure!  build file wasn't produced.
Please read INSTALL and associated NOTES files.  You may also have to look over
your available compiler tool chain or change your configuration.

***** Unsupported api compatibility level: 3.0
```

```
msp@msppc:~/openssl/openssl$ perl Configure linux-x86_64 --api=3

Failure!  build file wasn't produced.
Please read INSTALL and associated NOTES files.  You may also have to look over
your available compiler tool chain or change your configuration.

***** Unsupported api compatibility level: 3
```

cc @levitte

**Comment by levitte** [comment]
We do support deprecation at minor boundaries. *Removal* is restricted to major boundaries, however.

**Comment by piraces** [comment]
Hi, I will like to contribute and I think this is a good starting point for it.
Can I work on this bug and feature request?

**Comment by mspncp** [comment]
> Can I work on this bug and feature request?

Sure, every new contributor is welcome. I think @levitte has so many other things to do at the moment that he won't mind if you pick this issue. But of course if you need some advice or help during your work, don't hesitate to ask him, he's our Perl Wizard ;-).

**Comment by mattcaswell** [comment]
This was fixed by overridden by #10364. Closing.
